packages/app-mobile/components/NotesBar.tsx

```diff
@@ -0,0 +1,278 @@
+		</View>
+	);
+
+	const dividerComp = (
+		<View style={styles().divider}></View>
+	);
+
+	const closeButtonComp = (
+		<TouchableOpacity onPress={props.toggleNotesBar}>
+			<Icon name="close" style={[styles().top, styles().closeIcon]}/>
+		</TouchableOpacity>
+	);
+
+	function renderIconButton(icon: JSX.Element, onPress: ()=> Promise<void>) {
+		return (
+			<TouchableOpacity style={styles().button} activeOpacity={0.8} onPress={onPress}>{icon}</TouchableOpacity>
+		);
+	}
+
+
+	const handleNewNote = async (isTodo: boolean) => {
+		let folderId = props.selectedFolderId !== Folder.conflictFolderId() ? props.selectedFolderId : null;
+		if (!folderId) folderId = props.activeFolderId;
+
+		props.dispatch({
+			type: 'NAV_BACK',
+		});
+
+		const newNote = await Note.save({
+			parent_id: folderId,
+			is_todo: isTodo ? 1 : 0,
+		}, { provisional: true });
+
+		props.dispatch({
+			type: 'NAV_GO',
+			routeName: 'Note',
+			noteId: newNote.id,
+		});
+	};
+
+	const addNoteButtonComp = renderIconButton(<Icon name='document-text-outline' style={styles().buttonIcon} />, () => handleNewNote(false));
+	const addTodoButtonComp = renderIconButton(<Icon name='checkbox-outline' style={styles().buttonIcon} />, () => handleNewNote(true));
+
+	const topComp = (
+		<View>
+			<View style={[styles().topContainer, styles().horizontalFlex]}>
+				{titleComp}
+				{closeButtonComp}
+			</View>
+			{dividerComp}
+		</View>
+	);
+
+	const handleQuerySubmit = async () => {
+		if (!query) {
+			setNotes(props.items);
+			return;
+		}
+
+		let searchQuery = query.trim();
+
+		if (searchQuery === '') {
+			setNotes(props.items);
+			return;
+		}
+
+		searchQuery = searchQuery.toLowerCase();
+
+		// Find notes in folder using only note title
+		const result = props.items.filter(item => {
+			let noteTitle = Note.displayTitle(item);
+			noteTitle = noteTitle.toLowerCase();
+
+			return noteTitle.includes(searchQuery);
+		});
+
+		setNotes(result);
+	};
+
+	const searchInputComp = (
+		<View style={[styles().horizontalFlex, styles().searchInput]}>
+			<Icon name='search' style={[styles().top, styles().searchIcon]}/>
+			<TextInput style={[styles().top, styles().nativeInput]} placeholder='Search' onChangeText={setQuery} value={query} onSubmitEditing={handleQuerySubmit} placeholderTextColor={theme.dividerColor} />
+		</View>
+	);
+
+	const inputGroupComp = (
+		<View style={{ width: '100%' }}>
+			<View style={[styles().padding, styles().horizontalFlex, styles().inputGroup]}>
+				{searchInputComp}
+				{addNoteButtonComp}
+				{addTodoButtonComp}
+			</View>
+			{dividerComp}
+		</View>
+	);
+
+	let flatListRef: any = React.useRef(null);
+
+	const NotesBarListComp = (
+		<FlatList
+			data={notes}
+			renderItem={({ item }: { item: any }) => {
+				if (item.is_todo) {
+					return <NotesBarListItem note={item} todoCheckbox_change={props.todoCheckbox_change} />;
+				} else {
+					return <NotesBarListItem note={item} />;
+				}
+			}}
```

Comment:
Consider wrapping this and similar callbacks in a [`useCallback`](https://stackoverflow.com/a/58930853/17055750) hook.

We want to be able to update `query` without forcing a full re-render. Observing that `(() => {}) != (() => {})`, the `FlatList`'s `props` currently change for each re-render, thus forcing a full re-render of the `FlatList` when it might not be necessary.

Reply:
I'll do that, thanks a lot for this.